contracts/contracts/token/OUSD.sol

```diff
@@ -584,20 +609,111 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
             _totalSupply,
             _rebasingCredits,
             _rebasingCreditsPerToken
         );
     }
+
+    function delegateYield(address from, address to)
+        external
+        onlyGovernor
+    {
+        require(from != to, "Cannot delegate to self");
+        require(
+            yieldFrom[to] == address(0) &&
+                yieldTo[to] == address(0) &&
+                yieldFrom[from] == address(0) &&
+                yieldTo[from] == address(0),
+            "Blocked by existing yield delegation"
+        );
+        RebaseOptions stateFrom = rebaseState[from];
+        RebaseOptions stateTo = rebaseState[to];
+
+        require(
+            stateFrom == RebaseOptions.NotSet ||
+                stateFrom == RebaseOptions.StdNonRebasing ||
+                stateFrom == RebaseOptions.StdRebasing,
+            "Invalid rebaseState from"
+        );
+
+        require(
+            stateTo == RebaseOptions.NotSet ||
+                stateTo == RebaseOptions.StdNonRebasing ||
+                stateTo == RebaseOptions.StdRebasing,
+            "Invalid rebaseState to"
+        );
+
+
+        if (
+            alternativeCreditsPerToken[from] == 0 &&
+            (stateFrom == RebaseOptions.NotSet ||
+                stateFrom == RebaseOptions.StdRebasing)
+        ) {
+            _rebaseOptOut(from);
+        }
+        if (
+            alternativeCreditsPerToken[to] > 0 &&
+            (stateTo == RebaseOptions.NotSet ||
+                stateTo == RebaseOptions.StdNonRebasing)
+        ) {
+            _rebaseOptIn(to);
+        }
+
+        // Set up the bidirectional links
+        yieldTo[from] = to;
+        yieldFrom[to] = from;
+        rebaseState[from] = RebaseOptions.YieldDelegationSource;
+        rebaseState[to] = RebaseOptions.YieldDelegationTarget;
+
+        uint256 balance = balanceOf(from);
+        uint256 credits = _balanceToRebasingCredits(balance);
+
+        // Local
+        _creditBalances[from] = balance;
+        alternativeCreditsPerToken[from] = 1e18;
+        _creditBalances[to] += credits;
+
+        // Global
+        nonRebasingSupply -= balance;
+        _rebasingCredits += credits;
+    }
+
+    function undelegateYield(address from) external onlyGovernor {
+        // Require a delegation, which will also ensure a valid delegation
+        require(yieldTo[from] != address(0), "");
+
+        address to = yieldTo[from];
+        uint256 fromBalance = balanceOf(from);
+        uint256 toBalance = balanceOf(to);
+        uint256 toCreditsBefore = _creditBalances[to];
+        uint256 toNewCredits = _balanceToRebasingCredits(toBalance);
+
+        // Remove the bidirectional links
+        yieldFrom[yieldTo[from]] = address(0);
+        yieldTo[from] = address(0);
+        rebaseState[from] = RebaseOptions.StdNonRebasing;
+        rebaseState[to] = RebaseOptions.StdRebasing;
+
+        // Local
+        _creditBalances[from] = fromBalance;
+        alternativeCreditsPerToken[from] = 1e18;
```

Comment:
⚪️ We can remove these two writes to `alternativeCreditsPerToken` because we can know that they were correct  when the yield delegation was created.

For the same reason, we can also remove the set to `alternativeCreditsPerToken` in the `YieldDelegationSource` section of `_adjustAccount()`.